### PR TITLE
Common tests form submissions test

### PR DESCRIPTION
**Pull request by snopoke**
@benrudolph

### CODE REVIEW

File: corehq/form_processor/tests.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+from django.test import SimpleTestCase
+from corehq.form_processor.utils import to_generic
```
Comment (dimagimon) [inline corehq/form_processor/tests.py]: F401 'to_generic' imported but unused

File: corehq/form_processor/tests.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                # For some reason this is necessary for travis
+                app_label = "case"
+
+        case = CaseWithNewProperty(new_property='some property', bar='some other property').to_generic()
+        props = case.dynamic_case_properties()
+        self.assertEqual(2, len(props))
+        props_dict = dict(props)
+        self.assertEqual('some property', props_dict['new_property'])
+        self.assertEqual('some other property', props_dict['bar'])
+
```
Comment (dimagimon) [inline corehq/form_processor/tests.py]: W391 blank line at end of file

File: corehq/form_processor/generic.py [hunk 3]
```diff
@@ -82,6 +84,11 @@ class GenericXFormInstance(JsonObject):
     build_id = StringProperty()
     export_tag = DefaultProperty(name='#export_tag')
 
+    is_error = BooleanProperty(default=False)
+    is_duplicate = BooleanProperty(default=False)
+    is_deprecated = BooleanProperty(default=False)
+    is_archived = BooleanProperty(default=False)
```
Comment (benrudolph) [inline corehq/form_processor/generic.py]: i like this

File: corehq/form_processor/generic.py [hunk 4]
```diff
@@ -181,3 +189,28 @@ class GenericCommCareCase(JsonObject):
...
+    def reverse_indices(self):
+        from corehq.form_processor.interfaces import FormProcessorInterface
+        return FormProcessorInterface.get_reverse_indices(self.domain, self.id)
+
+    def dynamic_case_properties(self):
+        """(key, value) tuples sorted by key"""
+        from jsonobject.base import get_dynamic_properties
+        json = self.to_json()
+        wrapped_case = self
+        if type(self) != GenericCommCareCase:
```
Comment (benrudolph) [inline corehq/form_processor/generic.py]: when does this ever happen?

File: corehq/form_processor/generic.py [hunk 5]
```diff
@@ -181,3 +189,28 @@ class GenericCommCareCase(JsonObject):
...
+            wrapped_case = GenericCommCareCase.wrap(self._doc)
+
+        # should these be removed before converting to generic?
+        exclude = ['computed_modified_on_', 'computed_', 'doc_type', 'initial_processing_complete']
+        return sorted([
+            (key, json[key]) for key in get_dynamic_properties(wrapped_case)
+            if re.search(r'^[a-zA-Z]', key) and key not in exclude
```
Comment (benrudolph) [inline corehq/form_processor/generic.py]: is this just to make sure it doesn't start with an `_`? if so might be more readable like:
`if not key.startswith('_')`

File: corehq/form_processor/interfaces.py [hunk 6]
```diff
@@ -100,3 +161,49 @@ def process(xform):
...
+    def delete_all_cases(cls):
+        cls._delete_all(CommCareCase.get_db(), 'case/get_lite')
+
+    @classmethod
+    @unit_testing_only
+    def delete_all_xforms(cls):
+        cls._delete_all(XFormInstance.get_db(), 'couchforms/all_submissions_by_domain')
+
+    @classmethod
+    @unit_testing_only
```
Comment (benrudolph) [inline corehq/form_processor/interfaces.py]: :+1:

**Comment by benrudolph** [comment]
monstrous. test failure looks real, but easy to fix

File: corehq/form_processor/generic.py (hunk 4 above)
Comment (snopoke) [inline corehq/form_processor/generic.py]: No idea, maybe never, can add a soft_assert

File: corehq/form_processor/generic.py (hunk 5 above)
Comment (snopoke) [inline corehq/form_processor/generic.py]: I think `#` is also an option

File: corehq/apps/hqcase/tests/test_case_assigment.py [hunk 7]
```diff
@@ -78,13 +77,13 @@ def test_assign_to_group(self):
...
         self._make_tree()
-        assign_case(self.primary, group._id, include_subcases=True, include_parent_cases=True)
+        assign_case(self.primary.id, group._id, include_subcases=True, include_parent_cases=True)
         self._check_state(new_owner_id=group._id, expected_changed=self.all)
 
     def test_assign_noop(self):
         self._make_tree()
         num_forms = get_number_of_forms_in_all_domains()
-        res = assign_case(self.primary, self.original_owner._id, include_subcases=True, include_parent_cases=True)
+        res = assign_case(self.primary.id, self.original_owner._id, include_subcases=True, include_parent_cases=True)
```
Comment (dimagimon) [inline corehq/apps/hqcase/tests/test_case_assigment.py]: E501 line too long (117 > 115 characters)

File: corehq/form_processor/generic.py (hunk 4 above)
Comment (benrudolph) [inline corehq/form_processor/generic.py]: ah didn't realize this was taken from the Case model

**Comment by benrudolph** [comment]
:white_check_mark: let me know if you want to add more commits

**Comment by benrudolph** [comment]
else i'm good with merging

**Comment by snopoke** [comment]
woop

**Comment by czue** [comment]
haven't had a chance to look at this closely but awesome that it's moving forwards!

**Comment by snopoke** [comment]
Still quite a long way to go and possibly currently in a worse state than
before but hopefully getting to a point where we can clean it up more
easily.

On 13 October 2015 at 21:44, Cory Zue <notifications@github.com> wrote:

> haven't had a chance to look at this closely but awesome that it's moving
> forwards!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/8598#issuecomment-147828714>.
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa

**Pull request merged**